File: BeagleUI/Source/UI/ListView.swift
//
//  ListView.swift
//  BeagleUI
//
//  Created by Daniel Tes on 12/09/19.
//  Copyright © 2019 Daniel Tes. All rights reserved.
//

struct ListView: Widget {
    let rows: [Widget]?
    let remoteDataSource: String?
    let loadingState: Widget?
    let direction: ListDirection = .vertical
}

enum ListDirection {
    case vertical
    case horizontal
}

File: beagle-ui/src/main/java/br/com/zup/beagleui/engine/framework/ui/ListView.kt
package br.com.zup.beagleui.engine.framework.ui

import br.com.zup.beagleui.engine.framework.core.Widget

enum class ListDirection {
    VERTICAL,
    HORIZONTAL
}

data class ListView(
    val rows: List<Widget>? = null,
    val remoteDataSource: String? = null,
    val loadingState: Widget? = null,
    val direction: ListDirection = ListDirection.VERTICAL
) : Widget